git/src/lib.rs

```diff
@@ -1,45 +1,76 @@
 extern crate git2;
 
 use git2::{Error, Repository};
+use std::path::Path;
 
-/// Given an existing git repository, it will read the blob that the reference and the filename
-/// point to and return it as a String.
-pub fn cat_file(repo: &Repository, reference: &str, filename: &str) -> Result<Vec<u8>, Error> {
-    let reference = repo.find_reference(reference)?;
-
-    let tree = reference.peel_to_tree()?;
-
-    let path = std::path::Path::new(filename);
-    let te = tree.get_path(path)?;
+pub trait GitOps {
+    fn cat_file(&self, repo_path: &Path, reference: &str, filename: &str)
+        -> Result<Vec<u8>, Error>;
+}
 
-    repo.find_blob(te.id()).map(|x| x.content().to_owned())
+pub struct GitHelper;
```

Comment:
I wonder whether this should have another name or not seeing as it's public. Maybe like `LibGitOps`, which hints as to how it's doing it's work (which is the dual of the trait name `GitOps`, which hints that the ops need to do something `Git` related however they want)?